src/System.Collections/src/System/Collections/Generic/LinkedList.cs

```diff
@@ -0,0 +1,606 @@
+                {
+                    throw new ArgumentException(SR.Invalid_Array_Type);
+                }
+            }
+        }
+
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        [SuppressMessage("Microsoft.Performance", "CA1815:OverrideEqualsAndOperatorEqualsOnValueTypes", Justification = "not an expected scenario")]
+        public struct Enumerator : IEnumerator<T>, System.Collections.IEnumerator
+        {
+            private LinkedList<T> list;
+            private LinkedListNode<T> node;
+            private int version;
+            private T current;
+            private int index;
+
+            const string LinkedListName = "LinkedList";
+            const string CurrentValueName = "Current";
+            const string VersionName = "Version";
+            const string IndexName = "Index";
+
+            internal Enumerator(LinkedList<T> list)
+            {
+                this.list = list;
+                version = list.version;
+                node = list.head;
+                current = default(T);
+                index = 0;
+            }
+
+            public T Current
+            {
+                get { return current; }
+            }
+
+            object System.Collections.IEnumerator.Current
+            {
+                get
+                {
+                    if (index == 0 || (index == list.Count + 1))
+                    {
+                        throw new InvalidOperationException(SR.InvalidOperation_EnumOpCantHappen);
+                    }
+
+                    return current;
+                }
+            }
+
+            public bool MoveNext()
+            {
+                if (version != list.version)
+                {
+                    throw new InvalidOperationException(SR.InvalidOperation_EnumFailedVersion);
+                }
+
+                if (node == null)
+                {
+                    index = list.Count + 1;
+                    return false;
+                }
+
+                ++index;
+                current = node.item;
+                node = node.next;
+                if (node == list.head)
+                {
+                    node = null;
+                }
+                return true;
+            }
+
+            void System.Collections.IEnumerator.Reset()
+            {
+                if (version != list.version)
+                {
+                    throw new InvalidOperationException(SR.InvalidOperation_EnumFailedVersion);
+                }
+
+                current = default(T);
+                node = list.head;
+                index = 0;
+            }
+
+            public void Dispose()
+            {
+            }
+        }
+    }
+
+    // Note following class is not serializable since we customized the serialization of LinkedList. 
+    [System.Runtime.InteropServices.ComVisible(false)]
```

Comment:
Apparently the guidance has changed (for security transparency attributes, too).
Updated to remove all the ComVisibleAttributes and Security*Attributes.